### ISSUE TITLE
Write Docs with Different Widgets and Methods

### ISSUE THREAD

**Issue by sumeshir26** [tukaan/tukaan#48]
This will make it much easier to get started, I didn’t even know this had so many options without looking at the code!

**Comment by rdbende** [comment]
Yeah, it'd be good, I'm just too lazy. 🙄

**Comment by rdbende** [comment]
@sumeshir26 There's some initial documentation with GitHub pages here: https://tukaan.github.io/docs.
Currently there are only two pages, but I recommend you to read it, as it may also be useful for Tkinter users. Also, if you have an improvement or want to make a new page, feel free to contribute!

**Comment by sumeshir26** [comment]
On the home screen, Blocking us mistyped as Blicking

**Comment by sumeshir26** [comment]
Also, did'nt you write a basics article on this repos Github Wiki? You should add it

**Comment by rdbende** [comment]
I would really appreciate, if somebody could help in this 🤗

**Comment by rdbende** [comment]
Also, writing comprehensible \_\_doc\_\_strings in English is not my strong point.

**Comment by sumeshir26** [comment]
I would truly do it but recently I have become so busy☹️Maby in my vacations tho

**Comment by im-coder-lg** [comment]
I just saw this tbh. Could help in this if I could get hands-on with the features. You sure it can work on a Manjaro ARM build on a Pi 4B? I'm changing my devcenter completely from Windows.
Also, anyone have any ideas on a grammar checking tool in the command line? If it's on Pamac, I could check for errors and fix them. Just to get it perfected, I am a perfectionist, hence I don't leave little details out.

**Comment by im-coder-lg** [comment]
Maybe I'll just draft it out during a small break. This is the last school week and I am gonna be busy till the end of May, but that doesn't mean I don't get to test things. Will try Tukaan (first timers, so it's gonna be something!).
Maybe, just a tip: if we get creative, we could configure code snippets and use screenshots for outputs. My Linux machine would be honoured for this job, and so will I. @rdbende try making a Tukaan demo repo too, so every functions can be demoed(a new word?) there.

**Comment by rdbende** [comment]
> You sure it can work on a Manjaro ARM build on a Pi 4B? I'm changing my devcenter completely from Windows.

In theory everything should work just fine on Linux. Unfortunately that's not the case on Windows, 'cause I haven't compiled any binary extension on it. So Tukaan currently works only on Linux.

> @rdbende try making a Tukaan demo repo too, so every functions can be demoed(a new word?) there.

Yeye, I'm trying to add as much demo snippets to new PRs as I can, and some day I'll move them to an example repo. These code snippets can also help you learn a bit of Tukaan, although I know, they're not necessarily informative or useful.

**Comment by im-coder-lg** [comment]
Hmm... Doesn't matter if it's informative, just keep it in the open source mode, I'll help during May. I have an exam up and then holidays, however will be busy till May. I will charter some MD on Obsidian by today.

**Comment by im-coder-lg** [comment]
Also for the documentation site, I recommend that you use another Jekyll theme, Cayman sucks in that.

**Comment by im-coder-lg** [comment]
Also for the documentation site, I recommend that you use another Jekyll theme, Cayman sucks in that.

**Comment by im-coder-lg** [comment]
Plus point: I have a ague but nice idea. As we are talking, I just forked Tukaan's docs repo. Not to make huge text upgrades, but to add a really visually appealing Jekyll theme to it. Don't worry, I am in control there, but I just need some code snippets(well, list all the functions and we will add them slowly, could take a long time, but once done, enjoy!), some free time and a nice Jekyll theme.

**Comment by im-coder-lg** [comment]
@rdbende saw your invite to join the docs repo as a collaborator, thanks for that big opportunity. I'll upload my changes there in a new branch late afternoon.

**Comment by rdbende** [comment]
@im-coder-lg I saw, you use Just-the-Docs, I think it's a nice stuff, and I temporarily added it as a remote theme to `tukaan/docs`. I wanna have some customization on it, and I started experimenting with some visual identity [here](https://tukaan.github.io/) (turned out to be quite crappy :((
So, thank you for your work in you fork.

**Comment by im-coder-lg** [comment]
10-4, boss! I'll explore some more things, I need some help on a Favicon though, so if you provide that, I'll work on that, add it, customize the website and hopefully make it attractive. Is Tukaan ready for an alpha build? That way, we could gain traction to the website and the modules...

**Comment by im-coder-lg** [comment]
Hmm, the sentence "The New Experience" has a worse font. Any good font ideas?

**Comment by rdbende** [comment]
> Is Tukaan ready for an alpha build? That way, we could gain traction to the website and the modules...

Yeah, I wanna merge the Gtk theming before that so Tukaan can look native on Linux, but otherwise I think it's ready for an alpha release.

> Hmm, the sentence "The New Experience" has a worse font. Any good font ideas?

Will work on that today

**Comment by Moosems** [comment]
I just noticed this issue. Would you like me to look into helping with the docs so that we can add all the widgets to the docs? If not, I take no offense at all. I do not have nearly as much understanding in web development so I may not be the greatest fit but I know a decent bit (enough to create an outline)

**Comment by im-coder-lg** [comment]
Well, we do need extra help, so you can do it! No need to know web dev, all you need is Ruby, Bundler, Jekyll and some Markdown skill.

**Comment by Moosems** [comment]
Would you be able to give a quick rundown on those languages and the how to edit the site?

**Comment by im-coder-lg** [comment]
Well, you first need Ruby 2.7.3 on your Mac. Then, open your terminal with sudo perms, then make sure the gem command exists. Gem is the command call for Ruby's apckage manager. With that, run `gem install Jekyll bundler`. Then, to get the packages, go to the docs repo root and run `bundle install && bundle exec jekyll serve`. Then, you have a server running at localhost:4000/docs, and you can use Markdown to manage the pages. If you want to add config, then you should master YAML. That's basically Jekyll, a SSG.

**Comment by Moosems** [comment]
I've been trying for about an hour to update ruby but because I use a mac it is being a great pain 😩. I keep getting errors with SSL certificates and whatnot. I have Ruby 2.3.7 installed currently but am not sure if that will suffice. Does it require Ruby 2.7.3 or will my current version work? Thanks for the instructions by the way 🙏

**Comment by sumeshir26** [comment]
I guess 2.3 should work, but i dont have a mac so i haventrried

**Comment by sumeshir26** [comment]
And even if you write just the markdown and give it to us, we will add it and stuff

**Comment by Moosems** [comment]
Got it!

**Comment by Moosems** [comment]
@im-coder-lg I noticed that the docs pages are just markdown files. Those were what I was referring to when I meant help writing the docs. I could do that for y'all without ruby 😁

**Comment by im-coder-lg** [comment]
Look, listen to me. If you wanna preview the site, get Jekyll. If not, just use MD and get a CDN like Netlify to do test deployments.

You got Shortwave?

**Comment by Moosems** [comment]
I just got Shortwave. I also have Discord but can't use it until 4 cause of wifi restrictions

**Comment by Moosems** [comment]
I also created a Tukaan workspace

**Comment by Moosems** [comment]
Whats your Username so I can invite you?

**Comment by im-coder-lg** [comment]
Well, afaik, you need to add my email. But I would prefer Twitter to Shortwave. But anyway, can you add a ProtonMail account? I can't make Gmail because of the name restrictions. My name isn't offensive, but at the same time, I just can't afford my identity to be out in the world. I am just 13.

Maybe Twitter or Discord. Shortwave seems a bit Outta range since I use one lettered names for GitHub work.

**Comment by Moosems** [comment]
That makes sense. I don't have twitter (home rules) so that won't work. I'll add a ProtonMail account quickly. My account is moosems-coding@protonmail.com. I will delete this comment in 35 minutes so that should be enough time to add me

**Comment by Moosems** [comment]
Email me at moosems-coding@protonmail.com using ProtonMail

**Comment by Moosems** [comment]
@im-coder-lg I need to take down the email soon so that I don't start to get scammers

**Comment by Moosems** [comment]
I am not able to download and use Jekyll with my Ruby and because Apple locks down their products so hard my Mac doesn't want a Ruby update

**Comment by im-coder-lg** [comment]
Hmm.. try Gitpod services.

**Comment by insolor** [comment]
@Moosems technically you don't need Jekyll locally to contribute to the docs. You can just fork the repo, edit it, test on your account (you can enable Pages on your fork and see the result on the URL `<username>.github.io/docs`), then create a pull request to add your changes to the original repository.

Local Jekyll just simplifies testing and reduces time when you see results after you make changes.

**Comment by Moosems** [comment]
> @Moosems technically you don't need to have Jekyll locally to contribute to the docs. You can just fork the repo, edit it, test on your account (you can enable Pages on your fork and see the result on the URL `<username>.github.io/docs`), then create a pull request to add your changes to the original repository.
> 
> Local Jekyll just simplifies testing and reduces time when you see results after you make changes (just a few seconds instead of several minutes).

Sounds good! Is there anywhere I could go to get an explanation on Jekyll?

**Comment by im-coder-lg** [comment]
Well, the forum, Jekyll's official site, maybe even with people on the GitHub Pages team. But if you need some more help, I'll be there.

**Comment by Moosems** [comment]
Here's my idea for the doc creation:

I can make the actual markdown files and do the writing because English is my native language so it will be easier for me and I can write them pretty fast. Since I can't run Jekyll locally you can do the backend Jekyll stuff so that the work is divided according to resources and we could be most effective that way. If you guys want to do the markdown files that's fine but I can do it for you to take a bit of the workload off of y'all

**Comment by im-coder-lg** [comment]
I'll do the Jekyll backend stuff! My PC has Jekyll installed(that was my first project here, creating blogs via Jekyll), hence I can testt he website.

**Comment by Moosems** [comment]
Nice! Sounds like a plan!

**Comment by rdbende** [comment]
Yes, soon, because I've just rewritten the whole layouting thingy, and that will definitely need some explanation ;)

**Comment by Moosems** [comment]
Once the layouting is completed the docs should be completed pretty quickly

**Comment by Moosems** [comment]
> Yes, soon, because I've just rewritten the whole layouting thingy, and that will definitely need some explanation ;)

Am I correct that it will keep with the grid system?

**Comment by Moosems** [comment]
I think I'll have some time this week to work on updating the documentation

**Comment by Moosems** [comment]
I'll start powering through this sometime. @rdbende Can you get me a list of stables widgets that you would like documented and then I'll get to work?

**Comment by Moosems** [comment]
@rdbende What needs to be updated and what needs to be added?

**Comment by rdbende** [comment]
Basically we need a doc page on every widget, that hasn't been documented yet, and we have to update the docs about the `App` and `Window` classes (now there's a separate `App` object and `MainWindow` and `Window` widget).

**Comment by Moosems** [comment]
Wait really? Can you give a quick explanation?

**Comment by rdbende** [comment]
Yep. The idea behind it is to separate the concept of "The App" and the root window, it it makes it easier to display a splashscreen while the window loads, and make every widget appear instantly as the window becomes visible. It works like this:

```py
import tukaan

app = tukaan.App(name="My app", author="rdbende", version="2.0.0b4")  # These properties are later used to automatically fill the splash screen, the about dialog, and some other stuff with the needed information
# Nothing appears on the screen

splash = tukaan.SplashScreen(show_progressbar=True)


def setup(window):
    menubar = tukaan.MenuBar(window)

    some_widget = tukaan.SomeWidget(window)
    some_widget.grid()


win = tukaan.MainWindow(title="Main window of My app", setup=setup, on_load=splash.hide)

app.run()
```

**Comment by Moosems** [comment]
Oooooooh. Is this in the most recent Pypi release?

**Comment by rdbende** [comment]
No, it's only here on Github, and the MenuBar widget doesn't even exist (I'm working on it right now)

**Comment by Moosems** [comment]
I'm so excited for the Menubar to be honest because once the menubar and text widget are added I'm bugging out with tkinter and moving DIP onto tukaan. Will be annoying but hey, I want to help work more with Tukaan

**Comment by Moosems** [comment]
Oh, @rdbende I haven't seen you on ##learnpython in a while so I thought I'd ask you here: Can you help me with my linenumber widget? I'm trying to get dragging to work better and know you know tk better deep down. If you're up to the task let me know. no worries if not. In the meantime I will work on writing some more docs for Tukaan

**Comment by Moosems** [comment]
@rdbende I get an error when using the Splashscreen code when using the newest commit from the github repo

**Comment by rdbende** [comment]
Hmmm... 🤔

**Comment by Moosems** [comment]
Maybe its on your computer without being commited

**Comment by rdbende** [comment]
Whoops, that file is untracked, so I didn't commit it lol. Will fix it soon.

**Comment by Moosems** [comment]
So I noticed in the code that the comment said that the menubar would autofill but upon testing the code without the splashscreen it just uses the default tk one. Any thoughts?

**Comment by rdbende** [comment]
> the MenuBar widget doesn't even exist (I'm working on it right now)

The menubar stuff isn't finished yet.

**Comment by Moosems** [comment]
Ah

**Comment by Moosems** [comment]
So what have I missed recently?

**Comment by rdbende** [comment]
Anyone who would be interested in taking on this task? It would be greatly appreciated. The [new `main` branch](https://github.com/tukaan/tukaan/tree/main) is now quite stable.

**Comment by Moosems** [comment]
If you could give a very brief usage explanation for them and the possible args and Leary's I could take this on this summer.

**Comment by rdbende** [comment]
> Also, anyone have any ideas on a grammar checking tool in the command line? 

@im-coder-lg if you still want this, `aspell` is a really nice tool. `pacman -S aspell`

**Comment by im-coder-lg** [comment]
Dang I forgot about this! Sorry, I don't really know when I can help. I have a lot of exams coming up, but I will get holidays. If anybody can make a list of things to add for documentation, I can definitely help.

**Comment by rdbende** [comment]
@Moosems, @im-coder-lg
At some point now I will actually make a todo list :D

**Comment by Moosems** [comment]
Yay!!!!
